### DISCUSSION TITLE
Newbie - Clean Project

### DISCUSSION THREAD

**Discussion by jaybytez** [vercel/next.js#13468] [Help]
When switching between development branches, is there a type of "clean" that should be run, sorry coming from a Java/Maven world.  I understand when updating a project or switching branches of doing "npm install" for any new libraries added to the package.json, but what about files that may be in next directories after running a build and then updating the project and some of the js/jsx files have been removed.  Will the old files still be around or does the "npm run build" clean out build directories from previous builds?

**Comment by angad777** [comment]
I don't think that would be necessary, your version control should be able to manage this and switch back / revert files to the same state (including .next files) to a point in time on your local machine where they were untouched. Hope that makes sense. 

If you run into into issues, you can always stop / start the local dev server, just to be extra sure.

**Reply by jaybytez** [reply]
Thanks for the response.  That makes sense.  Wasn’t sure, also, if you use SSG and run a build and then during development delete jsx files, whether the old built files hang around or whether running npm build cleans them up.
